lib/pure/os.nim

```diff
@@ -2009,28 +2009,32 @@ proc removeFile*(file: string) {.rtl, extern: "nos$1", tags: [WriteDirEffect], n
   if not tryRemoveFile(file):
     raiseOSError(osLastError(), file)
 
-proc tryMoveFSObject(source, dest: string): bool {.noWeirdTarget.} =
+proc tryMoveFSObject(source, dest: string, isMoveDir: bool): bool {.noWeirdTarget.} =
   ## Moves a file or directory from `source` to `dest`.
   ##
   ## Returns false in case of `EXDEV` error.
   ## In case of other errors `OSError` is raised.
   ## Returns true in case of success.
+  const AccessDeniedError = OSErrorCode(5)
   when defined(windows):
     when useWinUnicode:
       let s = newWideCString(source)
       let d = newWideCString(dest)
-      if moveFileExW(s, d, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32: raiseOSError(osLastError(), $(source, dest))
+      result = moveFileExW(s, d, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) != 0'i32
     else:
-      if moveFileExA(source, dest, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32: raiseOSError(osLastError(), $(source, dest))
+      result = moveFileExA(source, dest, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) != 0'i32
   else:
-    if c_rename(source, dest) != 0'i32:
-      let err = osLastError()
-      if err == EXDEV.OSErrorCode:
-        return false
-      else:
-        # see whether `strerror(errno)` is redundant with what raiseOSError already shows
-        raiseOSError(err, $(source, dest, strerror(errno)))
-  return true
+    result = c_rename(source, dest) == 0'i32
+
+  if not result:
+    let err = osLastError()
+    var isAccessDeniedError: bool
+    when defined(windows):
+      isAccessDeniedError = isMoveDir and err == AccessDeniedError
+    else:
+      isAccessDeniedError = err == EXDEV.OSErrorCode
```

Comment:
```suggestion
    let isAccessDeniedError =
      when defined(windows): isMoveDir and err == AccessDeniedError
      else: err == EXDEV.OSErrorCode
```

Reply:
huh, didn't know when could be used that way just like an if. Nice :D